app/scripts/directives/oscSecrets.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscSecrets", function($uibModal, $timeout) {
+    return {
+      restrict: 'E',
+      scope: {
+        pickedSecrets: "=model",
+        allSecrets: '=',
+        namespace: "=",
+        additionalInputFieldName: "@?",
+        additionalInputFieldTooltip: "@?",
+        type: "@",
+        alertsCallback: '&',
```

Comment:
Let's just pass in the alerts array rather than adding alerts to alerts message service and having a callback. That's the pattern we've used elsewhere.